### PR TITLE
Link event calendar

### PR DESCRIPTION
**Pull request by Tikipiou**
# Link event calendar with 

## Overview

Linked event item from the calendar page to the event details page. And cleaning some code.

## Modifications

- Updated `Calendar.kt` for cleaning and design
- Updated `EventDisplay.kt` to redirect to `EventDetails.kt`
- Updated `Reminder.kt` for cleaning and design `EventDetails.kt`
- Created `InfiniteScrollableDaysList.kt` from `Calendar.kt` for clarity

## Testing

- Updated `CalendarTest.kt`
- Updated `ReminderTest.kt`

## Dependencies

- No additional dependencies required.

### CODE REVIEW

File: app/src/main/java/com/swent/assos/ui/screens/calendar/Reminder.kt [hunk 1]
```diff
@@ -134,13 +142,37 @@ fun Reminder(calendarViewModel: CalendarViewModel) {
                           Text(
                               modifier = Modifier.padding(top = 2.dp),
                               text =
-                                  "${event.second.startTime?.format(DateTimeFormatter.ofPattern("HH"))}" +
+                                  "${
+                                          event.second.startTime?.format(
```
Comment (Reza-rgb) [inline app/src/main/java/com/swent/assos/ui/screens/calendar/Reminder.kt:146]: why startTime would be null ?

File: app/src/main/java/com/swent/assos/ui/screens/calendar/Reminder.kt [hunk 2]
```diff
@@ -134,13 +142,37 @@ fun Reminder(calendarViewModel: CalendarViewModel) {
...
+                                              DateTimeFormatter.ofPattern(
+                                                  "HH"
+                                              )
+                                          )
+                                      }" +
                                       "." +
-                                      "${event.second.startTime?.format(DateTimeFormatter.ofPattern("mm"))}" +
+                                      "${
+                                                  event.second.startTime?.format(
```
Comment (Reza-rgb) [inline app/src/main/java/com/swent/assos/ui/screens/calendar/Reminder.kt:154]: why startTime would be null ?

File: app/src/main/java/com/swent/assos/ui/screens/calendar/Reminder.kt [hunk 3]
```diff
@@ -134,13 +142,37 @@ fun Reminder(calendarViewModel: CalendarViewModel) {
...
+                                                      DateTimeFormatter.ofPattern(
+                                                          "mm"
+                                                      )
+                                                  )
+                                              }" +
                                       " - " +
-                                      "${event.second.endTime?.format(DateTimeFormatter.ofPattern("HH"))}" +
+                                      "${
+                                                  event.second.endTime?.format(
```
Comment (Reza-rgb) [inline app/src/main/java/com/swent/assos/ui/screens/calendar/Reminder.kt:162]: why would endTime be null ?

File: app/src/main/java/com/swent/assos/ui/screens/calendar/Reminder.kt [hunk 4]
```diff
@@ -134,13 +142,37 @@ fun Reminder(calendarViewModel: CalendarViewModel) {
...
+                                                      DateTimeFormatter.ofPattern(
+                                                          "HH"
+                                                      )
+                                                  )
+                                              }" +
                                       "." +
-                                      "${event.second.endTime?.format(DateTimeFormatter.ofPattern("mm"))}",
+                                      "${
+                                                  event.second.endTime?.format(
```
Comment (Reza-rgb) [inline app/src/main/java/com/swent/assos/ui/screens/calendar/Reminder.kt:170]: why would endTime would be null ?

**Review by Reza-rgb — CHANGES_REQUESTED** [review]
Good PR ! You just have minor changes to solve. Some variables should not be null

File: app/src/main/java/com/swent/assos/ui/screens/calendar/Reminder.kt [hunk 5]
```diff
@@ -78,8 +81,8 @@ fun Reminder(calendarViewModel: CalendarViewModel) {
                 fontSize = 12.sp,
                 lineHeight = 26.sp,
                 fontFamily = FontFamily(Font(R.font.sf_pro_display_regular)),
-                fontWeight = FontWeight(400),
-                color = MaterialTheme.colorScheme.onBackground,
+                fontWeight = FontWeight.Light,
```
[the remark this answers is not in the document]
Comment (Tikipiou) [reply app/src/main/java/com/swent/assos/ui/screens/calendar/Reminder.kt:84]: It's better

File: app/src/main/java/com/swent/assos/ui/screens/calendar/Reminder.kt (hunk 1 above)
Comment (Tikipiou) [reply app/src/main/java/com/swent/assos/ui/screens/calendar/Reminder.kt:146]: We could change that later

File: app/src/main/java/com/swent/assos/ui/screens/calendar/Reminder.kt (hunk 3 above)
Comment (Tikipiou) [reply app/src/main/java/com/swent/assos/ui/screens/calendar/Reminder.kt:162]: Later

File: app/src/main/java/com/swent/assos/ui/screens/calendar/Reminder.kt (hunk 2 above)
Comment (Tikipiou) [reply app/src/main/java/com/swent/assos/ui/screens/calendar/Reminder.kt:154]: We could change that later

File: app/src/main/java/com/swent/assos/ui/screens/calendar/Reminder.kt (hunk 4 above)
Comment (Tikipiou) [reply app/src/main/java/com/swent/assos/ui/screens/calendar/Reminder.kt:170]: Later

**Review by Reza-rgb — APPROVED** [review]
All good !

**Pull request merged**